hazelcast/src/main/java/com/hazelcast/spi/impl/NodeEngineImpl.java

```diff
@@ -107,10 +109,23 @@ public NodeEngineImpl(Node node) {
                 operationService,
                 eventService,
                 wanReplicationService,
+                new ConnectionManagerPacketHandler(),
                 executionService);
         quorumService = new QuorumServiceImpl(this);
     }
 
+    class ConnectionManagerPacketHandler implements PacketHandler {
+        // ConnectionManager is only available after the NodeEngineImpl is available. This is a
+        // way around this problem.
+        @Override
+        public void handle(Packet packet) throws Exception {
+            ConnectionManager connectionManager = node.getConnectionManager();
```

Comment:
Once this PR is merged, I'll cast to TcpConnectionManager instead of checking for PacketHandler. I don't want to wait for another 1.5 hours build.